includes/class-sensei-course.php

```diff
@@ -3912,6 +3912,44 @@ public function get_days_to_completion_total() {
 
 		return (int) $days_to_completion;
 	}
+
+	/**
+	 * Get average days to completion for all courses filtered by courses.
+	 *
+	 * @since 4.5.0
+	 * @access public
+	 *
+	 * @param array $courses_ids Courses ids to filter by.
+	 * @return int Total days to completion, rounded to the highest integer.
+	 */
+	public function get_days_to_completion_total_filter_courses( array $courses_ids ) : int {
```

Comment:
I was thinking about this method that duplicates the old one and adds only one condition (`AND {$wpdb->comments}.comment_post_ID IN (%1s)`).

Two ideas came to my mind:
1. Add an optional param (`array $courses_ids = []`). It leaves us with a single method and removes duplication. However, I'm not fan of optional params, so I found another option:
2. We leave this method, but when we pass empty `$courses_ids` we don't add condition for filtering by `comment_post_ID`. Then, we call this method from the old one. Here, we remove duplication and have cleaner API without optional params :)

What do you think?

Reply:
1. The first I've implemented it with an optional parameter, but the first issue was that the way the original function was used was that it wants all the records, so how do we distinguish between an empty array sent and no parameter sent, since in the first case we should return an empty array and in the second we want to return ALL the elements. 

2. I didn't want to make it mandatory since it is a public function, so I wasn't sure if the developers would be using it. 


All in all, I like the idea of deprecating the function and making the parameter mandatory. 
if you both are ok with it, I would go with that one.

Reply:
Yes, deprecating works for me too 👍

Reply:
Addressed these comments in the comment above: https://github.com/Automattic/sensei/pull/5091/#discussion_r870638885